<a href="https://colab.research.google.com/github/Stesilva16/Eng-de-dados--/blob/main/while.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#1) Construam um algoritmo que leia 2 notas de 1000 alunos e diga se ele
#está ou não está aprovado com uma media 7.

if condição:
  bloco de código

while condição:
  bloco de código

execução normal do sistema

In [ ]:
numero = int(input('Digite um número positivo: '))

if numero <=0:
  print(f'O número digitado {numero} não é positivo!')
else:
  print(f'O número digitado {numero} é positivo!')

Digite um número positivo: -10
O número digitado -10 não é positivo!


In [ ]:
numero = int(input('Digite um número positivo: '))

while numero <= 0:
  numero = int(input(f'O número {numero} não é positivo! Favor digitar um número maior que 0'))

print('Você digitou um número válido!')



Digite um número positivo: -9
O número -9 não é positivo! Favor digitar um número maior que 06
Você digitou um número válido!


In [ ]:
#2) Faça um algoritmo que escreva na tela 1000x Bom dia!
contador = 1
while contador <= 500:
  print(f' {contador}')
  contador += 1

In [ ]:
#3) Faça um algoritmo para escrever na tela os números de 0 a 500
i = 0
while i <= 500:
  print(i, end=' ')
  i += 1
print('\n')
print(i)

0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 27

In [ ]:
#4) Faça um algoritmo para escrever na tela os números de 300 a 100 em ordem decrescente
i = 300
while i >= 100:
  print(i, end=' ')
  i -= 1

print('\n',i)

300 299 298 297 296 295 294 293 292 291 290 289 288 287 286 285 284 283 282 281 280 279 278 277 276 275 274 273 272 271 270 269 268 267 266 265 264 263 262 261 260 259 258 257 256 255 254 253 252 251 250 249 248 247 246 245 244 243 242 241 240 239 238 237 236 235 234 233 232 231 230 229 228 227 226 225 224 223 222 221 220 219 218 217 216 215 214 213 212 211 210 209 208 207 206 205 204 203 202 201 200 199 198 197 196 195 194 193 192 191 190 189 188 187 186 185 184 183 182 181 180 179 178 177 176 175 174 173 172 171 170 169 168 167 166 165 164 163 162 161 160 159 158 157 156 155 154 153 152 151 150 149 148 147 146 145 144 143 142 141 140 139 138 137 136 135 134 133 132 131 130 129 128 127 126 125 124 123 122 121 120 119 118 117 116 115 114 113 112 111 110 109 108 107 106 105 104 103 102 101 100 
 99


In [ ]:
#4) Faça um algoritmo para escrever na tela os números que são múltiplos de 4 de 300 a 100
# em ordem decrescente 
i = 300
while i >= 100:
  if i % 4 == 0:
    print(i)
  i -= 1

300
296
292
288
284
280
276
272
268
264
260
256
252
248
244
240
236
232
228
224
220
216
212
208
204
200
196
192
188
184
180
176
172
168
164
160
156
152
148
144
140
136
132
128
124
120
116
112
108
104
100


In [ ]:
#1) Faça um algoritmo que imprima na tela uma sequência de 10 números pares a partir do 1.
i = 2
while i <= 20:
  print(i, end=' ')
  i += 2

i = 1
while i <= 20:
  if i % 2 == 0:
    print(i, end='')
  i += 1

i = 1
while i <= 10:
  print(i * 2, end=' ')
  i += 1


2 4 6 8 10 12 14 16 18 20 

In [ ]:
#2) Faça um programa que apresente na tela os 10 primeiros números negativos.
i = -1
while i >= -10:
  print(i, end=' ')
  i -= 1

-1 -2 -3 -4 -5 -6 -7 -8 -9 -10 

In [ ]:
#3) Faça um algoritmo que leia um número positivo e imprima na tela todos os divisores desse número
numero = int(input('Digite um número qualquer: '))
while numero < 1:
  numero = int(input('Número inválido! Digite um número maior do que 0: '))

contador = 1
while contador <= numero:
  if numero % contador == 0:
    print(contador)
  contador += 1


Digite um número qualquer: 7
1
7


In [ ]:
#4) Faça um programa que leia N notas e mostre na tela a média das notas lidas.
qtde = int(input('Digite quantas notas serão digitadas: '))
while qtde < 1:
  qtde = int(input('Quantidade inválida! Digite uma quantidade maior que 0: '))

contador = 1
soma = 0
while contador <= qtde:
  nota = float(input('Digite a nota do aluno: '))
  while nota < 0 or nota > 10:
    nota = float(input('Nota inválida! Digite novamente a nota do aluno: '))
  soma += nota
  contador += 1

media = soma / qtde
print(f'A media das notas digitadas foi: {media:.2f}')

Digite quantas notas serão digitadas: 5
Digite a nota do aluno: -5
Nota inválida! Digite novamente a nota do aluno: 7
Digite a nota do aluno: 0
Digite a nota do aluno: 10
Digite a nota do aluno: -8
Nota inválida! Digite novamente a nota do aluno: 8
Digite a nota do aluno: 5
A media das notas digitadas foi: 6.00


In [ ]:
while True:
  nota = float(input('Digite a nota do aluno: Caso queira sair do programa, digite um valor negativo'))
  if nota < 0:
    break
  while nota > 10:
    nota = float(input('Nota inválida! Digite novamente a nota do aluno: '))

Digite a nota do aluno: Caso queira sair do programa, digite um valor negativo8
Digite a nota do aluno: Caso queira sair do programa, digite um valor negativo7
Digite a nota do aluno: Caso queira sair do programa, digite um valor negativo10
Digite a nota do aluno: Caso queira sair do programa, digite um valor negativo25
Nota inválida! Digite novamente a nota do aluno: 6
Digite a nota do aluno: Caso queira sair do programa, digite um valor negativo-1


In [ ]:
# 5) Escreva um algoritmo que leia um valor inicial A e uma razão R e 
#imprima uma seqüência em P.A. contendo 10 valores.
A = int(input('Digite a valor inicial da P.A: '))
R = int(input('Digite a Razão da P.A: '))

contador = 1
while contador <= 10:
  print(A, end=' ')
  A += R
  contador += 1

Digite a valor inicial da P.A: 3
Digite a Razão da P.A: 2
3 5 7 9 11 13 15 17 19 21 

In [ ]:
#6) Escrever um algoritmo que gera e escreve os números ímpares entre 100 e 200.
i = 101
while i <= 200:
  print(i, end=' ')
  i += 2

i = 100
while i <= 200:
  if i % 2 != 0:
    print(i, end=' ')
  i += 1

101 103 105 107 109 111 113 115 117 119 121 123 125 127 129 131 133 135 137 139 141 143 145 147 149 151 153 155 157 159 161 163 165 167 169 171 173 175 177 179 181 183 185 187 189 191 193 195 197 199 

In [ ]:
#Fatorial
num = int(input('Digite um número maior ou igual a 0: '))
while num < 0:
  num = int(input('Digite um número maior ou igual a 0: '))

contador = num - 1
fatorial = num

while contador >= 1:
  fatorial *= contador
  print(f'{num} x', end=' ')
  contador -= 1
  num -= 1
print('1', end=' ')  

if num < 2:
  print(f'{num}')
else:
  print(f'= {fatorial}')

Digite um número maior ou igual a 0: 5
5 x 4 x 3 x 2 x 1 1


In [ ]:
fatorial = int(input('Digite um número inteiro '))
while fatorial < 0:
  fatorial = int(input('Número inválido. Digite um número inteiro '))  
total = 1
print(f'{fatorial}! =',end=' ')
while fatorial >= 1:
  total *= fatorial
  print(f'{fatorial}x',end='')
  fatorial-=1

print(f' = {total}')

Digite um número inteiro 5
5! = 5x4x3x2x1x = 120


In [ ]:
#TABUADA
i = 1
j = 0

while i < 11:
  while j < 11:
    print(f'{i} X {j} = {i * j}')
    j += 1
  i += 1
  j = 0
  print('\n')

1 X 0 = 0
1 X 1 = 1
1 X 2 = 2
1 X 3 = 3
1 X 4 = 4
1 X 5 = 5
1 X 6 = 6
1 X 7 = 7
1 X 8 = 8
1 X 9 = 9
1 X 10 = 10


2 X 0 = 0
2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18
2 X 10 = 20


3 X 0 = 0
3 X 1 = 3
3 X 2 = 6
3 X 3 = 9
3 X 4 = 12
3 X 5 = 15
3 X 6 = 18
3 X 7 = 21
3 X 8 = 24
3 X 9 = 27
3 X 10 = 30


4 X 0 = 0
4 X 1 = 4
4 X 2 = 8
4 X 3 = 12
4 X 4 = 16
4 X 5 = 20
4 X 6 = 24
4 X 7 = 28
4 X 8 = 32
4 X 9 = 36
4 X 10 = 40


5 X 0 = 0
5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45
5 X 10 = 50


6 X 0 = 0
6 X 1 = 6
6 X 2 = 12
6 X 3 = 18
6 X 4 = 24
6 X 5 = 30
6 X 6 = 36
6 X 7 = 42
6 X 8 = 48
6 X 9 = 54
6 X 10 = 60


7 X 0 = 0
7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63
7 X 10 = 70


8 X 0 = 0
8 X 1 = 8
8 X 2 = 16
8 X 3 = 24
8 X 4 = 32
8 X 5 = 40
8 X 6 = 48
8 X 7 = 56
8 X 8 = 64
8 X 9 = 72
8 X 10 = 80


9 X 0 = 0
9 X 1 = 9
9 X 2 = 18
9 X 3 =

1) Faça um algoritmo que imprima na tela uma sequência de 10 números pares a partir do 1.

2) Faça um programa que apresente na tela os 10 primeiros números negativos.

3) Faça um algoritmo que leia um número positivo e imprima na tela todos os divisores desse número

4) Faça um programa que leia N notas e mostre na tela a média das notas lidas.

5) Escreva um algoritmo que leia um valor inicial A e uma razão R e imprima uma seqüência em P.A. contendo 10 valores.

6) Escrever um algoritmo que gera e escreve os números ímpares entre 100 e 200.

7) Crie uma tabuada do 1 ao 10 utilizando o comando while. Ex de saída: 1 x 0 = 0 1 x 1 = 1 1 x 2 = … 1 x 10 = 10

2 x 0 = 0 2 x 1 = 2 … 2 x 10 = 20

….

10 x 0 = 0 10 x 1 = 10 … 10 x 10 = 100

8) Desenvolver um algoritmo que leia um número não determinado de valores e calcule e escreva a média aritmética dos valores lidos, a quantidade de valores positivos, a quantidade de valores negativos e o percentual de valores negativos e positivos.

9) Escrever um algoritmo que leia uma quantidade desconhecida de números e conte quantos deles estão nos seguintes intervalos: [0-25], [26-50], [51-75] e [76-100]. A entrada de dados deve terminar quando for lido um número negativo.

10) Escreva um algoritmo que leia um valor inicial A e imprima a seqüência de valores do cálculo de A! e o seu resultado. Ex: 5! = 5 X 4 X 3 X 2 X 1 = 120

In [ ]:
qtd_pos = 0
qtd_neg = 0
perc_pos = 0
perc_neg = 0
soma = 0
total = 0

while True:
  n = int(input('Digite um número: '))
  if n == 0:
    break
  if n > 0:
    qtd_pos += 1
  else:
    qtd_neg += 1
  soma += n
  total += 1

if total == 0:
  print('Não foi digitado nenhum número!')
else:
  print(f'A média dos números lidos foi de {soma/total}')
  print(f'A quantidade de valores positivos é {qtd_pos}')
  print(f'A quantidade de valores negativos é {qtd_neg}')
  print(f'O percentual de valores positivos é {(qtd_pos/total) * 100}%')
  print(f'O percentual de valores negativos é {(qtd_neg/total) * 100}%')


Digite um número: 10
Digite um número: -5
Digite um número: 8
Digite um número: -3
Digite um número: 0
A média dos números lidos foi de 2.5
A quantidade de valores positivos é 2
A quantidade de valores negativos é 2
O percentual de valores positivos é 50.0
O percentual de valores negativos é 50.0


In [ ]:
intervalo1 = 0
intervalo2 = 0
intervalo3 = 0
intervalo4 = 0

while True:
  n = int(input('Digite um número qualquer: '))
  if n < 0:
    print('Você digitou um número negativo e encerrou o programa!')
    break
  if 0 <= numero <= 25:
    intervalo1 += 1
  if 26 <= numero <= 50:
    intervalo2 += 1
  if 51 <= numero <= 75:
    intervalo3 += 1
  if 76 <= numero <= 100:
    intervalo4 += 1

print(f'Qtd de números no intervalo [0-25]{intervalo1}')
print(f'Qtd de números no intervalo [26-50]{intervalo2}')
print(f'Qtd de números no intervalo [51-75]{intervalo3}')
print(f'Qtd de números no intervalo [76-100]{intervalo4}')


Digite um número qualquer: 10
Digite um número qualquer: 5
Digite um número qualquer: 0
Digite um número qualquer: -1
Você digitou um número negativo e encerrou o programa!
3
0
0
0


In [ ]:
if aula 'acabou':
  break

2
4
6
8
10
12
14
16
18
20
